### DISCUSSION TITLE
Removing Functionalities

### DISCUSSION THREAD

**Discussion by Cernelius** [triplea-game/triplea#12343] [General]
With reference to:
https://github.com/triplea-game/triplea/issues/12333

Was that functionality removed without even asking in forum? Am I missing something?

Anyway, my suggestion is never removing any functionalities without leaving a discussion about it open in forum for at least 1 year (not every map-maker checks forum constantly) and never opening such discussion for any functionalities added less than 5 years ago (giving at least some years for map-makers to use new functionalities).

If agreed, this would need documentation as guide-lines for developers about removing functionalities.

**What other functionalities have been removed like that?**

---

Though I would agree that "the_pact_of_steel" should be complete as a guide.

I think having added this functionality without having documented it in "the_pact_of_steel" has been sloppy if @WCSumpton is right there is no example in it.

My suggestion (in line with what I just said) is that, if a functionality is found non-used by any maps comprising the_pact_of_steel, a developer should give an ultimatum in forum for anyone to add it to this map and removing the functionality if nobody does it within 1 year thereafter (no matter if the functionality has been added recently).

This should actually never happen if it is agreed that every game-related functionality should always be documented in the "the_pact_of_steel" game (especially new additions).

---

I remember that the fuctionality was added by @ron-murhammer (and I consider it a good addition even though I've not used it so far, but I intend to).

**Comment by WCSumpton** [comment]
@Cernelius
Maybe this discussion could be opened on the main form.

Cheers...
